### ISSUE TITLE
Add a compact mode to the OSC 

### ISSUE THREAD

**Issue by houssemko** [Samillion/ModernZ#208]
### Expected behavior of the wanted feature

By moving the remaining time and total time in front and in the end of the progress bar , the OSC can be more compact

**Comment by houssemko** [comment]
I understand , that's great

**Comment by Samillion** [comment]
Re-opening, to keep track. Once it's implemented, you'll be notified with the relative PR.

**Comment by houssemko** [comment]
Okay , thanks

**Comment by xiaoleichen** [comment]
I added a compact mode: 

https://github.com/xiaoleichen/ModernZ/commit/3ab8a8d4cecaacaaad31a8c20c734351a5374752

It's not perfect given the dynamic time code with showing/hiding hour part and ms part. It adjusts seekbar width to accommodate ms part, but not the hour part. It just reserve fixed space for hour part.

It ignores show_chapter_title and always hides the chapter title. But it adds a new chapters button for the same functionality: hovering on the button shows current chapter and clicking on it brings up chapter selection.

It also adjusts position of media title, but I usually hide media title for the minimalist UI with compact mode.

<img width="1728" alt="Screenshot 2024-12-09 at 11 34 58 AM" src="https://github.com/user-attachments/assets/58628528-3767-45d0-b5bb-2a7b8b58afa4">

**Comment by Samillion** [comment]
What do you guys think about this layout?

Also solves #264 

**Changes:**
- Reduce OSC footprint by re-organizing elements
- Move `chapter_title` below title
- Move `time_stamp` and `cache_info`

![Untitled](https://github.com/user-attachments/assets/6e03cd4a-d417-4cdc-b610-d7fa87a99ce2)

**Comment by houssemko** [comment]
I think there's still room for the jump buttons , you could make it optional to remove them ?

**Comment by Samillion** [comment]
Sorry, I forgot to clarify. This is the same layout with all the options, just with a reduced footprint.

No options have been removed.

**Comment by houssemko** [comment]
the 10s jup buttons are missing though

**Comment by Samillion** [comment]
That's just my config.

![Untitled1](https://github.com/user-attachments/assets/93e3500c-51b7-42ef-81aa-dedbc335ef8e)

**Comment by houssemko** [comment]
that's perfect , i likr it.

**Comment by Samillion** [comment]
Test script: [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua)
Commit: https://github.com/Samillion/ModernZ/commit/058fb6ec65b7b2616dc36734a98f4dd795606ace

Please test and let me know if there are any issues or changes that need to be made.

**Changes:**
- Reduced `bottomhover_zone` to 130
- Reduced `time_font_size` to 16
- Reduced `raise_subtitle_amount` to 125
- Reduced `playpause_size` to 28
- Reduced `fade_alpha` to 130
- Reduced possible OSC height to 140 and 100 (dependent on title function)
- Moved `chapter_title` below title
- Moved time stamps next to left side buttons
- Moved `cache_info` next to right side buttons
- Improved elements visibility based on window width

Pinging @Keith94 and @Xurdejl as they have shown interest in compact mode.

**Comment by da3dsoul1** [comment]
Wow, compact mode is so cool! Thanks!

**Comment by Keith94** [comment]
>Please test and let me know if there are any issues or changes that need to be made.

If I may make a few suggestions.

1. The volume slider is looking a bit huge. I suggest using a width of around 55 instead of 80.
2. If above change is made, the time can be moved to the left which creates more room.
3. Chapter title should have more padding vertically and increased font size.
4. Buttons look too close to the seekbar.. should be centered better vertically. Distance between seekbar and play/pause is 26px. Play/pause to bottom window border is 32px.
5. Personally I'd move the chapter title down, right beside the time and remove support for mouse commands which would also solve hitbox issues. Only issue would be long chapter titles would have to be truncated with ellipses so it would fit in the given space (if that's possible with Lua).

**Comment by Samillion** [comment]
**Updated:** [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua)
**Commit:** https://github.com/Samillion/ModernZ/commit/2ee659c532bec569b2443c0c7cd8bef94f66d721

**Changes:**
- Increase `chapter_title` font size and padding
- Position `title` based on the existence of `chapters` to eliminate empty space with no chapters
- Move side and middle buttons to be more symmetrically centered
- Decrease volume bar width to 55

With `chapter_title`:
![image](https://github.com/user-attachments/assets/7e23a66d-acf9-4af7-87d1-6f7a1da00145)

Without `chapter_title`:
![image](https://github.com/user-attachments/assets/62fd9aab-d3f1-4206-9175-aacf323840c8)


> If above change is made, the time can be moved to the left which creates more room.

Wouldn't this break "standard"? Most OSCs (local and online players) have `[vol_icon] [vol_bar] [time]` or move the volume bar all the way to the right.

I'm not rejecting this specific suggestion, I'm just concerned on user experience.

Do you think we should instead make the volume bar a user option? ie: show volume icon only if need be.

**Comment by Keith94** [comment]
>Wouldn't this break "standard"? Most OSCs (local and online players) have [vol_icon] [vol_bar] [time] or move the volume bar all the way to the right

Yep sorry that was my bad. I meant to say the margin could just be shifted over but that was an obvious change after all lol.

>Do you think we should instead make the volume bar a user option? ie: show volume icon only if need be.

Also yes should be an option. Any chance then you add a tooltip to the icon to show current percentage?

**Comment by Samillion** [comment]
> Also yes should be an option. Any chance then you add a tooltip to the icon to show current percentage?

Only with a monthly subscription that shows limited ads, and a premium option that increases in price every 5 months!

Just kidding lol. Definitely can do, will add this later as a separate feature.

About `chapter_title`, is the font size and padding better? I was thinking of adding a prefix to chapter_title, like

```
[title]
[prefix] [chapter]
```

Since they are connected to each other, if that makes sense.

**Possible options:**
- ↳
- ⤷
- ➥
- ↬
- ↘
- ╰─
- ⮞
-

**Comment by Keith94** [comment]
I think size is perfect but padding could be better. I know it's technically supposed to be compact but it looks crammed in lol. Maybe I'm comparing too much to SmartTube app's padding.

Option 2 would be good.

**Comment by Samillion** [comment]
Updated script. (Same link)
Commit: https://github.com/Samillion/ModernZ/commit/52b77e6ea4f6a220901668fcbe344092abd32ff1

**Changes:**
- Increase `chapter_title` padding
- Add `↳`

I tested them, that prefix is the only one that is kind of visible lol. Not sure if I should keep it or not, to be honest.

![image](https://github.com/user-attachments/assets/10a80aad-e09c-4d48-9356-29d0f48e717c)

**Comment by Keith94** [comment]
I'm AFK but yeah, it looks a bit tiny but also distracting in a way. I would remove it.

Can you also align the left margins for title and chapter. In https://github.com/Samillion/ModernZ/issues/208#issuecomment-2562913073 the D and N looked mismatched.

**Comment by Samillion** [comment]
> Can you also align the left margins for title and chapter. In [#208 (comment)](https://github.com/Samillion/ModernZ/issues/208#issuecomment-2562913073) the D and N looked mismatched.

True, it's off by 1 pixel.

![image](https://github.com/user-attachments/assets/76d1bb76-f8f8-4a75-ac92-3cc2cb89dc68)

**Comment by Samillion** [comment]
**Updated:** [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua)
**Commit:** https://github.com/Samillion/ModernZ/commit/c75e35e59652f58639bb672110be155a4c0e40ff

**Changes:**
- Fix `chapter_title` 1 pixel offset
- Improve chapter check for `title` position by checking `user_opts.show_chapter_title` as well
- Remove `chapter_title` prefix

**Comment by Samillion** [comment]
Hmmm. Should the seek bar be lowered by 5 ish pixels? (along with title and chapter title)

**Comment by da3dsoul1** [comment]
![image](https://github.com/user-attachments/assets/b4c74deb-8121-45a2-95b6-af8dac390e31)
I think there should be a space

**Comment by Samillion** [comment]
Weird, by any chance have you increased font size for time stamps in your config? `time_font_size`

Hmmm, maybe spacing should account for font size. But... but....... math 😭

**Comment by da3dsoul1** [comment]
> time_font_size

I have it on 18

**Comment by Samillion** [comment]
Yeah, when I was moving the elements, I measured it for 16. I'll squeeze the two cells in my brain to base spacing on font size for time stamps.

**Comment by xiaoleichen** [comment]
I wonder how do people feel about this type of layout?

1. Seekbar is moved on the same level as timestamps. Its length is adjusted dynamically depending whether to show the hour part and / or millisecond part of the timestamps.
2. A chapter selection button is added. Hovering shows current chapter and clicking brings up chapter menu. (need a better icon).
3. cache info is moved at the same level as title and made bigger (18). Title width is shortened to make space for cache info.
<img width="1308" alt="mpv" src="https://github.com/user-attachments/assets/b1ada68c-bc1d-40c0-a7f0-190554ecf9d7" />

**Comment by Keith94** [comment]
>Seekbar is moved on the same level as timestamps.

Not a huge fan of it especially how the seekbar shrinks or expands whenever the time width changes. Looks a bit jarring to my eye.

>A chapter selection button is added. Hovering shows current chapter and clicking brings up chapter menu. (need a better icon).

Probably wouldn't use it. Can right- or middle-click the chapter skip buttons to achieve basically the same thing. What kind of icon would be suitable for that? 

>cache info is moved at the same level as title and made bigger (18). Title width is shortened to make space for cache info.

Torn on this one. Could be suitable for something like vertical videos where the UI is already squished a lot, but when the UI gets too small I think elements start disappearing anyway.

**Comment by xiaoleichen** [comment]
Makes sense.

One clarification on the seekbar shrinking/expanding: it doesn't do that on its own as video plays. It does that only when you click the time stamp on the left to show / hide ms part. To be honest I don't see the point of showing MS part at all...

The dependence on the hour part is determined when video is opened: it reserves more spaces only if the total length is over 1 hour, regardless whether the hour part actually shows during video playback.

**Comment by Keith94** [comment]
I see. That isn't as bad as I thought it was. Thanks for clarifying.

Still some folk may find it a bit overcomplicated, but who can say for sure.

**Comment by Keith94** [comment]
@Samillion

Would it look weird if you displayed cache speed on a new line? Essentially add a line break after cache fill and remove the separator.

**Comment by Samillion** [comment]
> I wonder how do people feel about this type of layout (updated with cache info)?

I absolutely love what you did, great work.

>     1. Seekbar is moved on the same level as timestamps. Its length is adjusted dynamically depending whether to show the hour part and / or millisecond part of the timestamps.

My concern is same as @Keith94. Though maybe it's possible to reserve empty space for timestamps (like what's done with cached time) so that the seekbar width stays static.

>     2. A chapter selection button is added. Hovering shows current chapter and clicking brings up chapter menu. (need a better icon).

I going to add a Menu button once it's merged in mpv: https://github.com/mpv-player/mpv/pull/15499

This will basically be an all in one button for those that want to disable all or some buttons, and replace it with one.

>     3. cache info is moved at the same level as title and made bigger (18). Title width is shortened to make space for cache info.

That's exactly how I did in the beginning, but unfortunately it wasn't well received, user experience wise.

P.S.
Really like the no chapter markers options you added, well done.

Let me know if you'll maintain that fork, and I'll gladly link to it in the main repo.

**Comment by Samillion** [comment]
> Would it look weird if you displayed cache speed on a new line? Essentially add a line break after cache fill and remove the separator.

Ah, interesting idea. I'll try it out and get back to ya.

**Comment by Keith94** [comment]
When title and chapter title are both disabled, OSC height should be more like 85 not 100. I guess that would need additional function sadly.. more bloat but useful to have

**Comment by Samillion** [comment]
@da3dsoul1

> > time_font_size
> 
> I have it on 18

I have zero confidence in my maths skills lol. Would love if you guys confirm that it's working properly.

**Updated:** [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua)
**Commit:** https://github.com/Samillion/ModernZ/commit/562b2640b4356c943991aa849bc0bdd541f90e3b

**Changes:**
- Correctly position time codes based on `user_opts.time_font_size` with an incremental `x` offset.

```lua
    local function time_pos_x(current_font_size)
        local offset, x
        local base_font_size = 16

        if current_font_size >= base_font_size then
            local difference = current_font_size - base_font_size
            x = (state.tc_ms and 1 or 0) + (show_hours and 1 or 0)
            offset = difference * (x + 2.5 + difference / 10) -- incremental increase
        else
            local difference = base_font_size - current_font_size
            x = (state.tc_ms and 2.5 or 0) + (show_hours and 1 or 0)
            offset = -difference * (x + 2.5 + difference / 10) -- incremental decrease
        end

        return offset
    end

    local x_offset = time_pos_x(user_opts.time_font_size) or 0
```

**Comment by Samillion** [comment]
> When title and chapter title are both disabled, OSC height should be more like 85 not 100. I guess that would need additional function sadly.. more bloat but useful to have

Shouldn't be a problem at all, will address that once the time code thing is done.

**Comment by xiaoleichen** [comment]
> @da3dsoul1
> 
> > > time_font_size
> > 
> > 
> > I have it on 18
> 
> I have zero confidence in my maths skills lol. Would love if you guys confirm that it's working properly.
> 
> **Updated:** [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua) **Commit:** [562b264](https://github.com/Samillion/ModernZ/commit/562b2640b4356c943991aa849bc0bdd541f90e3b)
> 
> **Changes:**
> 
> * Correctly position time codes based on `user_opts.time_font_size` with an incremental `x` offset.
> 
> ```lua
>     local function time_pos_x(current_font_size)
>         local offset, x
>         local base_font_size = 16
> 
>         if current_font_size >= base_font_size then
>             local difference = current_font_size - base_font_size
>             x = (state.tc_ms and 1 or 0) + (show_hours and 1 or 0)
>             offset = difference * (x + 2.5 + difference / 10) -- incremental increase
>         else
>             local difference = base_font_size - current_font_size
>             x = (state.tc_ms and 2.5 or 0) + (show_hours and 1 or 0)
>             offset = -difference * (x + 2.5 + difference / 10) -- incremental decrease
>         end
> 
>         return offset
>     end
> 
>     local x_offset = time_pos_x(user_opts.time_font_size) or 0
> ```

This doesn't seem to account for font family. The default font (mpv-osd-symbols) uses font from mpv which is `sans-serif` by default on mpv. On linux `sans-serif` matches `DejaVu Sans` which is a fatter font. The position is a bit off:

![Screenshot from 2024-12-27 02-54-04](https://github.com/user-attachments/assets/c3dede52-9575-4a84-87d7-01f8e3a7a041)

I wonder if tc_left and tc_right can be combined into one element. But I guess it wouldn't be possible to handle clicks separately.

Another issue if the cache info, which doesn't seems to adjust its position based on font size. Larger font cause overlay with right buttons:

![Screenshot from 2024-12-27 02-54-33](https://github.com/user-attachments/assets/933e18da-9b55-44df-907b-27a6f2efae6d)

**Comment by Samillion** [comment]
> This doesn't seem to account for font family. The default font (mpv-osd-symbols) uses font from mpv which is sans-serif by default on mpv. On linux sans-serif matches DejaVu Sans which is a fatter font. The position is a bit off
>
> I wonder if tc_left and tc_right can be combined into one element. But I guess it wouldn't be possible to handle clicks separately.

Indeed, the only solution would be to combine them into a single element. I've been really considering doing that, then change the click behavior to:
- Left click: display remaining time
- Right click: show milliseconds (ms)

> Another issue if the cache info, which doesn't seems to adjust its position based on font size. Larger font cause overlay with right buttons

This should be solved soon once I implement the two lines idea. I'll reserve space for speed so it doesn't change position.

**Comment by Samillion** [comment]
Yup, combining them both into a single element with left/right click behavior made it so much better.

**Comment by Samillion** [comment]
@Keith94 I like the two lines idea. I implemented it like this:

![image](https://github.com/user-attachments/assets/de7ee69d-5cd9-4d10-87d2-48013bfa79d3)

With a tooltip on hover that shows `Cached Time` and `Cache Speed`, respectively.

If cache_info_speed is disabled, the position of cached time is changed to be centered with the buttons.

Would this be a good implementation?

I removed the Cache: label because it looked weird to be honest.

**Comment by Samillion** [comment]
**Updated:** [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua)
**Commit:** https://github.com/Samillion/ModernZ/commit/83313579feb2befdf4129844bc994065b0730974

**Changes:**
- Combined time codes into one element for consistency `time_codes`
- Changed click functions to time code
- Display `cache_info` and `cache_info_speed` in two separate lines
- Display `cache_info` correctly if `cache_info_speed` is disabled
- Remove `Cache:` label and use tool tips on hover instead
- Display current volume on hover with `vol_ctrl` icon

![Untitled](https://github.com/user-attachments/assets/1cc753bb-82ba-463f-98d2-d5deed207280)

**Comment by Samillion** [comment]
Note to self:
Reverse time code click functions. Milliseconds should be on right click.

**Comment by Keith94** [comment]
Wicked job. Thanks for the changes. 2 minor things:

1. For local videos the cache speed isn't centered vertically. (Is such info useful for local videos?...hmm)
2. The height and width hit box for the new timecodes seem a tiny bit off. Needs more height and a little less width.

**Comment by xiaoleichen** [comment]
This is looking better and better.

The distance between cache info and the right buttons keeps changing as cached time and speed change. Since it's on the right side, would it be better to align to the right, such that it grows / shrinks with the space on the left and the distance to the button is always the same? This also helps when a bigger font is used.

<img width="274" alt="Screenshot 2024-12-27 135305" src="https://github.com/user-attachments/assets/09ab7914-3146-4686-b770-2d1bb6d8bc07" />

**Comment by xiaoleichen** [comment]
Another suggestion: since it's using `tc_left / tc_right` format for time codes, can we make the `tc_right` always display total time and toggle `tc_left` for time played and time remaining?

btw this is how Plex app does it:

* Mobile Plex app displays time code on left right side of the seek bar, so `tc_left` always shows time played, and `tc_right` shows total time or time remaining.
<img width="1169" alt="Screenshot 2024-12-27 at 5 53 12 PM" src="https://github.com/user-attachments/assets/5a4a495e-aa7e-4d14-90c3-4acd557bbe7d" />

* Desktop Plex app uses `tc_left / tc_right` format, so `tc_left` toggles between time played and time remaining, and `tc_right` always show total time.
 
![Screenshot 2024-12-27 174458](https://github.com/user-attachments/assets/385278b3-c45f-4162-a073-ddfb7d367710)

**Comment by Samillion** [comment]
> For local videos the cache speed isn't centered vertically. (Is such info useful for local videos?...hmm)

I asked the same question on the feature request for cache, and they said to enable for all types. I also only use cache speed for online videos only, so I enable it with my Stream auto profile

> The height and width hit box for the new timecodes seem a tiny bit off. Needs more height and a little less width.

Will try to fix as best as I can.

> The distance between cache info and the right buttons keeps changing as cached time and speed change. Since it's on the right side, would it be better to align to the right,

Great idea. I'll use `an=6` (align right), looks much better that way.

![image](https://github.com/user-attachments/assets/e15e08f3-3e9c-46fb-a8be-8d47e92be010)

> Another suggestion: since it's using tc_left / tc_right format for time codes, can we make the tc_right always display total time and toggle tc_left for time played and time remaining?

The mpv users would crucify me if I did that lol. We can maybe introduce a feature in the future to customize time display behavior

**Comment by Samillion** [comment]
@Keith94 
> When title and chapter title are both disabled, OSC height should be more like 85 not 100. I guess that would need additional function sadly.. more bloat but useful to have

This should suffice, right?

`osc_geo.h`:
- If both `no_title` and `no_chapter` are true, the height is `85`.
- If `no_title` is true but `no_chapter` is false, the height is `100`.
- If `no_title` is false, the height defaults to `140`.

```lua
    local no_title = 
        ((user_opts.title_mbtn_left_command == "" or user_opts.title_mbtn_left_command == "ignore") and
        (user_opts.title_mbtn_right_command == "" or user_opts.title_mbtn_right_command == "ignore")) or 
        not user_opts.show_title

    local no_chapter = 
        ((user_opts.chapter_title_mbtn_left_command == "" or user_opts.chapter_title_mbtn_left_command == "ignore") and
        (user_opts.chapter_title_mbtn_right_command == "" or user_opts.chapter_title_mbtn_right_command == "ignore")) or 
        not user_opts.show_chapter_title

    local osc_geo = {
        w = osc_param.playresx,
        h = no_title and (no_chapter and 85 or 100) or 140
    }
```

**Comment by Samillion** [comment]
**Updated:** [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua)
**Commit:** https://github.com/Samillion/ModernZ/commit/138e263431692b26036a37b584636867fe15f1a6

**Changes:**
- Decrease total height of OSC to `130`
- Dynamically decrease height of OSC based on `title` or `chapter_title` existence/function with `osc_height_offset`
- Improve `time_codes` hitbox height and width
- Improve `cache_info` and `cache_info_speed` position and alignment
- Reverse `time_codes` click functions

**Comment by Keith94** [comment]
Everything looks great. Except I'm a little confused. When cache info is enabled why does the cache time show up invisible on local videos?

![image](https://github.com/user-attachments/assets/98a1ee34-1927-4cfc-8661-df8ce324bbd7)

**Comment by Samillion** [comment]
Hmmm. I'm unable to re-produce that behavior, only reason it wouldn't show is because it's not caching anything. Any errors or indicators in your log about it?

![image](https://github.com/user-attachments/assets/282f6aed-b445-4a71-ab38-f132114b6d75)

**Comment by Samillion** [comment]
Ah, I think I know why.

```lua
        if not (cache_state and cache_state["seekable-ranges"] and
            #cache_state["seekable-ranges"] > 0) then
            -- probably not a network stream
            return ""
        end
```

**Comment by Samillion** [comment]
**Updated:** [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua)
**Commit:** https://github.com/Samillion/ModernZ/commit/87478e5ba91460d5665756a7f0fbdb4739587d8e

**Changes:**
- Display `cache_info` and `cache_info_speed` correctly according to `cache_state`

**Comment by Keith94** [comment]
I didn't really see anything related to cache but it's still happening. Weird.

[newlog.txt](https://github.com/user-attachments/files/18264987/newlog.txt)

**Comment by Samillion** [comment]
That is really strange. I'll try to re-produce the behavior, can't see anything indicating a problem in your log except for:

```
[   0.243][w][modernz] script-opts: unknown key sekbarbg_color, ignoring
```

Which isn't about cache.

**Comment by Samillion** [comment]
So cached time is not showing at all for you on local videos?

![image](https://github.com/user-attachments/assets/711f0da7-690c-4f6a-ad04-26177335c17e)

**Comment by Keith94** [comment]
>So cached time is not showing at all for you on local videos?

Yeah, not at all.. It used to show just the speed properly and cache time was invisible. Commit https://github.com/Samillion/ModernZ/commit/87478e5ba91460d5665756a7f0fbdb4739587d8e made it so speed doesn't show up now. Feels like I made a simple mistake somewhere lol.

**Comment by Samillion** [comment]
> Yeah, not at all.. It used to show just the speed properly and cache time was invisible. Commit [87478e5](https://github.com/Samillion/ModernZ/commit/87478e5ba91460d5665756a7f0fbdb4739587d8e) made it so speed doesn't show up now. Feels like I made a simple mistake somewhere lol.

Hmmm. Could it be related to your cache options in `mpv.conf`? Mine is:
```
### Cache
cache=yes
demuxer-max-bytes=400MiB
demuxer-max-back-bytes=200MiB
```

**Comment by Keith94** [comment]
Ahh, thank you. You found my simple mistake.

I left mpv's cache option alone which defaults to auto. When I set it to yes I can see the cache finally.

I guess if you set yours to "auto" or "no" then you should be able to reproduce that tooltip over invisible cache thingy. Hopefully.

**Comment by Samillion** [comment]
> I guess if you set yours to "auto" or "no" then you should be able to reproduce that tooltip over invisible cache thingy. Hopefully.

Good point. Tooltip should only show if cache is shown.

**Comment by Samillion** [comment]
**Updated:** [compact_osc/modernz.lua](https://github.com/Samillion/ModernZ/blob/dev_compact_osc_feat/modernz.lua)
**Commit:** https://github.com/Samillion/ModernZ/commit/245127d16f300db05be702fee4c30d1ca4d1a577

**Changes:**
- Only display cache tooltip if cache ranges exist.

@Keith94 I learned something new because of this bug. If you want to display `cache_info` for online only, then in `mpv.conf` set `cache=auto`

**Comment by Keith94** [comment]
That's great! 😎 I feel like modernz should follow that same setting, if possible. Don't show cache for local files unless you specifically have it on `yes`.
